internal/tui/components/dialog/custom_commands.go

```diff
@@ -0,0 +1,153 @@
+package dialog
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/opencode-ai/opencode/internal/config"
+	"github.com/opencode-ai/opencode/internal/tui/util"
+)
+
+// Command prefix constants
+const (
+	UserCommandPrefix    = "user:"
+	ProjectCommandPrefix = "project:"
+)
+
+// LoadCustomCommands loads custom commands from both XDG_CONFIG_HOME and project data directory
+func LoadCustomCommands() ([]Command, error) {
+	cfg := config.Get()
+	if cfg == nil {
+		return nil, fmt.Errorf("config not loaded")
+	}
+
+	var commands []Command
+
+	// Load user commands from XDG_CONFIG_HOME/opencode/commands
+	xdgConfigHome := os.Getenv("XDG_CONFIG_HOME")
```

Comment:
can we also make it possible to add commands in `$HOME/.opencode/commands` this way this behaves similar to the config path defaults.

Reply:
@kujtimiihoxha updated